bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/listener/DoorbirdEvent.java

```diff
@@ -233,11 +238,62 @@ private void decryptV1(ByteBuffer bb, String password5) throws IndexOutOfBoundsE
             logger.trace("Decryption FAILED");
             return;
         }
+        getFieldsFromDecryptedText(m, mLen);
+    }
+
+    private void decryptV2(ByteBuffer bb, @Nullable String v2DecryptionKey)
+            throws IndexOutOfBoundsException, BufferUnderflowException {
+        LazySodiumJava sodium = getLazySodiumJavaInstance();
+        if (sodium == null) {
+            logger.debug("Unable to decrypt v2 event because libsodium is not loaded");
+            return;
+        }
+
+        if (v2DecryptionKey == null) {
+            logger.debug("Unable to decrypt v2 event because decryption key is null");
+            return;
+        }
+
+        logger.debug("Decrypting v2 event, size of buffer: {}", bb.capacity());
+
+        // Get nonce and ciphertext arrays
+        bb.get(nonce, 0, nonce.length);
+        bb.get(ciphertext, 0, ciphertext.length);
+
+        // Set up the variables for the decryption algorithm
+        byte[] m = new byte[30];
+        long[] mLen = new long[30];
+        byte[] nSec = null;
+        byte[] c = ciphertext;
+        long cLen = ciphertext.length;
+        byte[] ad = null;
+        long adLen = 0;
+        byte[] nPub = nonce;
+        byte[] k = v2DecryptionKey.getBytes();
+
+        // Decrypt the ciphertext
+        logger.trace("Call cryptoAeadChaCha20Poly1305Decrypt with ciphertext='{}', nonce='{}', key='{}'",
+                HexUtils.bytesToHex(ciphertext, " "), HexUtils.bytesToHex(nonce, " "), HexUtils.bytesToHex(k, " "));
+        boolean success = sodium.cryptoAeadChaCha20Poly1305Decrypt(m, mLen, nSec, c, cLen, ad, adLen, nPub, k);
+        if (!success) {
+            /*
+             * Don't log at debug level since the decryption will fail for events encrypted with
+             * passwords other than the password contained in the thing configuration (reference API
+             * documentation for details)
+             */
+            logger.trace("Decryption FAILED");
```

Comment:
```suggestion
            logger.debug("Decryption FAILED");
```

Reply:
I left this as TRACE, because a failure here is "normal". The comment directly above explains why.

Reply:
Ok, as you carefully chosen this to be trace, it is fine by me.
I was somewhat triggered because earlier in this method there is a 
```
logger.debug("Decrypting v2 event, size of buffer: {}", bb.capacity());
``` 
and later on there is only a follow up in trace mode.

Reply:
Yes, I understand. Thanks. Given the wonky nature of how this works, it wasn't possible to have clear indication of an actual failure.